app/models/search.rb

```diff
@@ -12,27 +12,13 @@ class Search
   def date_from=(value)
     return if value.nil?
 
-    year = value[1]
-    month = value[2]
-    day = value[3]
-    @date_from = begin
-      Date.new(year, month, day)
-    rescue TypeError, Date::Error
-      value
-    end
+    build_date(value, "from")
   end
 
   def date_to=(value)
     return if value.nil?
 
-    year = value[1]
-    month = value[2]
-    day = value[3]
-    @date_to = begin
-      Date.new(year, month, day)
-    rescue TypeError, Date::Error
-      value
-    end
+    build_date(value, "to")
   end
 
   def results
```

Comment:
Maybe worth adding a test for what was failing to protect against a future regression.

Reply:
I've added tests for date range searches which do and do not return results.